EDU-4263: Minor clarifications to action billing usage w.r.t. Workflows (#3497)
temporalio/documentation: docs/evaluate/temporal-cloud/pricing.mdx

- **Workflow started**.
  Occurs via client start, client [Signal-With-Start](/sending-messages#signal-with-start), [Continue-As-New](/workflows#continue-as-new), or [Child Workflow](/child-workflows) start.

>   If a Workflow start attempt via client results in the server returning an error, an Action is not recorded. 

Reviewer: ```suggestion
  An Action also won't be recorded when a Workflow start attempt through a Client results in the server returning an error. 
```

**CHILD WORKFLOWS**

> - **Start Child Workflow Execution Initiated**.

Reviewer: ```suggestion
- **Start Child Workflow Execution initiated**.
```

>   The Parent Workflow durably recording the intent to start a Child Workflow results in one Action.

Reviewer: ```suggestion
  When the parent Workflow durably records the intent to start a Child Workflow, it results in one Action.
```

- **Child Workflow Execution attempted**.

>   The server's attempt to start the Child Workflow results in one Action.

Reviewer: ```suggestion
  When the server attempts to start the Child Workflow, it results in one Action.
```
